setup.py

```diff
@@ -9,15 +9,21 @@
 
 * `documentation <https://wsgi-kerberos.readthedocs.org/en/latest/>`_
 * `development version
-  <http://github.com/mkomitee/wsgi-kerberos/zipball/master#egg=wsgi-kerberos-dev>`_
+  <http://github.com/deshaw/wsgi-kerberos/zipball/master#egg=wsgi-kerberos-dev>`_
 
 """
 
+import os
+import re
 from setuptools import setup
 
+lib = os.path.join(os.path.dirname(__file__), "wsgi_kerberos.py")
+with open(lib) as fh:
+    version = re.search(r"""__version__ = ["'](.*?)["']""", fh.read()).group(1)
```

Comment:
Worth adding a unit test that the module provides a `__version__` attribute?

Reply:
Eh...the `__version__` attribute is authoritative, so I don't really think it's necessary. The package won't even install without it set.

Reply:
The point would be to catch that regression before it ever landed in the master branch, rather than waiting until a release to find out.

Also, with the current implementation, if `__version__` were removed, installation would fail with an unfriendly `AttributeError: 'NoneType' object has no attribute 'group'` error.

So what I was suggesting was:
- merely append `__version__` to the `from wsgi_kerberos import ...` line in `test_wsgi_kerberos.py`, so that if the attribute were ever removed, the tests would fail with an ImportError, blocking the associated PR from being merged
- ideally rewrite the `re.search(...).group(...)` in `setup.py` to give a clearer error message when it fails, or possibly even continue installing with a dummy version and just warn that version could not be detected. That way, if a bad change did get pushed straight to the master branch bypassing CI in our PR status checks (the project's GitHub settings may not forbid landing changes outside PRs), and this bug did make it into a release, it would either fail with a better error message or not even be a fatal error.

Up to you!

Reply:
I still don't think it's necessary. I've never seen unit tests for this in other projects. The code example for exacting the version I copied from flask, but I've seen similar examples in other projects.

Reply:
I guess the principle is that if it's public API, it deserves test coverage, but I get if that feels like overkill.

Just in case it's helpful to see some examples though, the popular `coverage` project has tests for its public `version_info` attribute:
https://github.com/nedbat/coveragepy/blob/34b665620889f5389178d0f37e8061c8ffa5759c/tests/test_version.py#L19 

I also test for the public `__version__` attribute (and similar package metadata) [in the unit tests for my `bidict` project](https://github.com/jab/bidict/blob/a38d51e4b559be0b82faefdfa6677788cfcd92fb/tests/test_metadata.py).

I don't see a "resolve conversation" button, but fine with me if you prefer to leave this as-is. :)